plugins/catalog-backend/src/service/CatalogPlugin.ts

```diff
@@ -59,6 +60,14 @@ class CatalogProcessingExtensionPointImpl
     this.#placeholderResolvers[key] = resolver;
   }
 
+  addPermissionRules(
```

Comment:
I'm not sure about this method here, as permissions have nothing to do with the catalog processor. Maybe at this point `CatalogProcessingExtensionPoint` needs to be renamed? 🤔

Reply:
In that case, perhaps a separate extension point (say `CatalogPermissionExtensionPoint`) containing this method can be created instead?

Reply:
Hi @vinzscam, would my above suggestion be workable or is this something that would require a more in-depth discussion with the team? Thanks again for your help with this issue.

Reply:
hi @lshwayne96, yeah your idea sounds good! please make the change :)

Reply:
Hi @nuritizra, I have made the requested changes, for your review please. Thank you!